native/src/utils/DatabaseConnector.ts

```diff
@@ -881,7 +889,13 @@ class DatabaseConnector {
   }
 
   async writeFile(path: string, data: string): Promise<void> {
-    return BlobUtil.fs.writeFile(path, data, 'utf8')
+    // Create directory structure if it doesn't exist
+    const dirname = path.substring(0, path.lastIndexOf('/'))
+    const dirExists = await RNFS.exists(dirname)
+    if (!dirExists) {
+      await RNFS.mkdir(dirname, { NSURLIsExcludedFromBackupKey: true })
+    }
```

Comment:
🔧 `mkdir` does not fail if the directory already exists ([docs](https://github.com/birdofpreyru/react-native-fs?tab=readme-ov-file#mkdir)), so you can just do:

```suggestion
    const dirname = path.substring(0, path.lastIndexOf('/'))
    await RNFS.mkdir(dirname, { NSURLIsExcludedFromBackupKey: true })
```